---
title: "figure5"
author: "Mette Ludwig"
date: "2022-01-06"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---


```{r}
library(DESeq2)
library(dplyr)
library(tidyr)
library(plyr)
library(forcats)
library(ggplot2)
library(ggpubr)
library(ggrepel)
library(VennDiagram)
library(stringr)
library(gprofiler2)
library(openxlsx)
library(grid)
library(rlist)
library(openxlsx)
library(wesanderson)
```

# Load RNA-seq data
```{r}
# Mapping data 
mapping.data <- read.table(gzfile("/projects/mludwig/Ludwig-2021/data/gene_info/Mus_musculus.GRCm38.90.gene_name_version2ensembl.txt.gz"), header = T)

# Bulk data 
path.meta <- "/projects/mludwig/Petersen-2022/data/bulk/jona-seq2_meta.csv"
path.counts <- "/projects/mludwig/Petersen-2022/data/bulk/jona-seq2_counts.csv"

bulk.meta <- read.csv(path.meta, sep = ",", header = T)
colnames(bulk.meta) <- gsub("\\.\\.", "\\.", colnames(bulk.meta))
bulk.meta$condition <- gsub(" ", "", bulk.meta$Condition1)
bulk.meta$sample <- gsub("0149_", "hypo", bulk.meta$SampleID)
bulk.meta$user.id <- bulk.meta$Subject.Name.User.selected.ID.
bulk.meta$cage <- bulk.meta$Cage.no..1.2.3.
bulk.meta$cage.recoded <- bulk.meta$cage_recoded
bulk.meta$RIN <- bulk.meta$RIN.value
bulk.meta$name <- paste(bulk.meta$sample, bulk.meta$condition, sep = "_")
bulk.meta <- bulk.meta[,c("name", "sample", "condition", "cage.recoded", "cage", "RIN", "user.id")]

bulk.counts <- read.csv(path.counts, sep = ",", header = T)

# Get gene IDs
bulk.counts[,1] <- gsub("\\..*", "", bulk.counts[,1])
bulk.counts[,1] <- mapping.data$gene_name_optimal[match(bulk.counts[,1], 
                                                   mapping.data$ensembl_gene_id)]

bulk.counts <- bulk.counts[!(is.na(bulk.counts[,1])),]
rownames(bulk.counts) <- bulk.counts[,1]
bulk.counts <- bulk.counts[,-c(1:6)]

colnames(bulk.counts) <- gsub("X0149_", "hypo", colnames(bulk.counts))
bulk.counts <- bulk.counts[, paste0("hypo", seq(1:ncol(bulk.counts)))]


# Get treatment groups
colnames(bulk.counts) <- paste(colnames(bulk.counts), 
                               bulk.meta$condition[match(colnames(bulk.counts), 
                                                         bulk.meta$sample)], sep = "_")
```


# Compute differentially expressed genes
```{r}
# Peptide vs. vehicle
counts.sub <- bulk.counts[, grep("vehicle|peptide", colnames(bulk.counts))]
condition <- gsub(".*_", "", colnames(counts.sub))
condition <- factor(condition, levels = c("vehicle", "peptide"))
cage <- factor(bulk.meta$cage.recoded[match(colnames(counts.sub), bulk.meta$name)])
RIN <- bulk.meta$RIN[match(colnames(counts.sub), bulk.meta$name)]
RIN <- scale(RIN)
design <- data.frame(condition = condition, cage = cage, RIN = RIN)

dds <- DESeqDataSetFromMatrix(counts.sub, DataFrame(design), ~ condition)
dds <- DESeq(dds)
DEGs.peptide <- results(dds, name="condition_peptide_vs_vehicle") 
DEGs.peptide <- data.frame(DEGs.peptide)

# Conjugate2 vs. vehicle
counts.sub <- bulk.counts[, grep("vehicle|conjugate2", colnames(bulk.counts))]
condition <- gsub(".*_", "", colnames(counts.sub))
condition <- factor(condition, levels = c("vehicle", "conjugate2"))
cage <- factor(bulk.meta$cage.recoded[match(colnames(counts.sub), bulk.meta$name)])
RIN <- bulk.meta$RIN[match(colnames(counts.sub), bulk.meta$name)]
RIN <- scale(RIN)
design <- data.frame(condition = condition, cage = cage, RIN = RIN)

dds <- DESeqDataSetFromMatrix(counts.sub, DataFrame(design), ~ condition)
dds <- DESeq(dds)
DEGs.conjugate2 <- results(dds, name="condition_conjugate2_vs_vehicle") 
DEGs.conjugate2 <- data.frame(DEGs.conjugate2)

# Conjugate1 vs. vehicle 
counts.sub <- bulk.counts[, grep("vehicle|conjugate1", colnames(bulk.counts))]
condition <- gsub(".*_", "", colnames(counts.sub))
condition <- factor(condition, levels = c("vehicle", "conjugate1"))
cage <- factor(bulk.meta$cage.recoded[match(colnames(counts.sub), bulk.meta$name)])
RIN <- bulk.meta$RIN[match(colnames(counts.sub), bulk.meta$name)]
RIN <- scale(RIN)
design <- data.frame(condition = condition, cage = cage, RIN = RIN)

dds <- DESeqDataSetFromMatrix(counts.sub, DataFrame(design), ~ condition)
dds <- DESeq(dds)
DEGs.conjugate1 <- results(dds, name="condition_conjugate1_vs_vehicle") 
DEGs.conjugate1 <- data.frame(DEGs.conjugate1)
```

# Figure 5a
```{r}
# Construct venn diagram
DEGs.peptide.venn <- DEGs.peptide[which(DEGs.peptide$padj < 0.05),]
DEGs.conjugate1.venn <- DEGs.conjugate1[which(DEGs.conjugate1$padj < 0.05),]

genes <- union(rownames(DEGs.peptide.venn), rownames(DEGs.conjugate1.venn))
venn.data <- data.frame(matrix(0, nrow = length(genes), ncol = 2))
rownames(venn.data) <- genes
colnames(venn.data) <- c("peptide", "conjugate1")

venn.data$peptide[which(rownames(venn.data) %in% rownames(DEGs.peptide.venn))] <- 1
venn.data$conjugate1[which(rownames(venn.data) %in% rownames(DEGs.conjugate1.venn))] <- 1

pal <- wes_palette("Darjeeling2", 4, type = "discrete")[2:3]

grid.newpage()
png("/projects/mludwig/Petersen-2022/output/figures/figure5a.png",
    pointsize = 1,
    width = 960, height = 960, res = 2250)

draw.pairwise.venn(area1 = nrow(subset(venn.data, peptide == 1)), 
                  area2 = nrow(subset(venn.data, conjugate1 == 1)), 
                  cross.area = nrow(subset(venn.data, peptide == 1 & conjugate1 == 1)),
                  category = c("Peptide", "Conjugate1"), 
                  lty = "blank",
                  fontfamily = "helvetica",
                  cat.fontfamily = "helvetica",
                  cat.fontface = "bold",
                  fill = c(pal[[1]], pal[[2]]),
                  margin = 0.05)
dev.off()
```


# Figure 5d
```{r}
# Compute differentially expressed genes for for conjugate1 vs. peptide
counts.sub <- bulk.counts[, grep("peptide|conjugate1", colnames(bulk.counts))]
condition <- gsub(".*_", "", colnames(counts.sub))
condition <- factor(condition, levels = c("peptide", "conjugate1"))
cage <- factor(bulk.meta$cage.recoded[which(bulk.meta$condition %in% condition)])
RIN <- bulk.meta$RIN[match(colnames(counts.sub), bulk.meta$name)]
RIN <- scale(RIN)
design <- data.frame(condition = condition, cage = cage, RIN = RIN)

dds <- DESeqDataSetFromMatrix(counts.sub, DataFrame(design), ~ condition)
dds <- DESeq(dds)
DEGs <- results(dds, name="condition_conjugate1_vs_peptide") 
DEGs <- data.frame(DEGs)

# Save output
DEGs.save <- DEGs[!(is.na(DEGs$padj)), ]
DEGs.save <- DEGs.save[order(DEGs.save$pvalue),]
DEGs.save <- cbind(gene = rownames(DEGs.save), DEGs.save)

write.xlsx(DEGs.save, 
           file = "/projects/mludwig/Petersen-2022/output/DEGs/DEGs_conjugate1_vs_semaglutide.xlsx", 
           rowNames = F)


# Construct volcano plot
volcano <- DEGs[!(is.na(DEGs$padj)), ]
volcano$label <- ""
volcano$label[min_rank(volcano$padj) <= 15] <- rownames(volcano)[min_rank(volcano$padj) <= 15]
volcano$col <- 4
volcano$col[which(volcano$padj <= 0.05 & abs(volcano$log2FoldChange) >= 0.5)] <- 1
volcano$col[which(volcano$padj <= 0.05 & abs(volcano$log2FoldChange) < 0.5)] <- 2
volcano$col[which(volcano$padj > 0.05 & abs(volcano$log2FoldChange) >= 0.5)] <- 3
volcano <- data.frame(volcano[, c("log2FoldChange", "padj", "col", "label")])

# Plot
volcano.plot <- ggplot(volcano, aes(y=(-log10(padj)), x=log2FoldChange, 
                                    fill=factor(col), label=label)) +
  xlab(expression(bold(paste(Log[2], " fold-change")))) + 
  ylab(expression(paste(bold(-log[10]),bold("("), bolditalic("P"),
                                   bold(")")))) + 
  geom_point(shape = 21, size = 3, alpha = 1) + 
  geom_hline(yintercept = -log10(0.05), linetype = "dashed") +
  geom_vline(xintercept = c(-.5, .5), linetype = "dashed") + 
  geom_text_repel(fontface="bold", size=3, color = "black") + theme_pubr() + 
      theme(legend.position = "none",
            axis.title = element_text(size=10, face="bold"),
            axis.text = element_text(size=10, face="bold"))  + 
      scale_fill_manual(values = c("1" = "#960000", "2" = "#CC9999", 
                                   "3" = "grey40", "4" = "grey80")) 
volcano.plot

ggsave("/projects/mludwig/Petersen-2022/output/figures/figure5d.png", 
       volcano.plot, height = 60, width = 90,
       units = "mm")
```

# Figure 5c
```{r}
# Identify top GO terms
query <- rownames(DEGs)[which(DEGs$padj < 0.05)]

module.GO.terms <- gost(query = query,
                        organism = "mmusculus",
                        ordered_query = F, correction_method = "fdr",
                        exclude_iea = T, 
                        sources = c("GO:BP", "GO:MF", "GO:CC", "KEGG", "REAC"),
                        evcodes = T)

GO.terms <- module.GO.terms[[1]] %>% select(term_name, term_size, p_value, intersection_size, intersection, term_id) %>% 
  filter(term_size <= 500) %>% arrange(p_value) %>% top_n(5, -p_value) %>%
  mutate(x = fct_reorder(str_wrap(term_name, 20), -p_value)) %>% 
  mutate(y = -log10(p_value)) 

# Save 
write.xlsx(GO.terms[, !(colnames(GO.terms) == "intersection")], 
           file = "/projects/mludwig/Petersen-2022/output/GO_terms/GO_terms_conjugate1_vs_semaglutide.xlsx", rowNames = F)

# Plot
GO.plot <- ggplot(GO.terms, aes(x,y)) + 
  geom_col(colour="black", width=1, fill="#D69C4E", alpha=0.5, size=0.3) +
  theme_pubr(legend="none") + 
  theme(axis.text.x = element_text(size=8, face="bold"),
        axis.text.y = element_text(size=8, face="bold"),
        axis.title.x = element_text(size=8, face="bold"),
        # margin: top, right, bottom, and left
        plot.margin = unit(c(0, 0.5, 0, 0), "cm")) +
  ggsci::scale_fill_lancet() +
  coord_flip() +
  xlab(NULL) + ylab(expression(paste(bold(-log[10]),bold("("),
                                       bolditalic("P"),bold(")")))) 
GO.plot


ggsave("/projects/mludwig/Petersen-2022/output/figures/figure5c.png", 
       GO.plot, height = 42, width = 70,
       units = "mm")
```


# Figure 5b
```{r}
# Identify differentially expressed genes intersecting with top GO terms
GO.genes <- lapply(GO.terms$intersection, noquote)
GO.genes <- lapply(GO.genes, function(x) str_split(x, pattern = ","))
GO.genes <- unique(unlist(GO.genes))

top.genes <- GO.genes[order(DEGs[GO.genes, "padj"], decreasing = F)][1:20]

GO.genes.data <- DEGs[top.genes, ]
GO.genes.data$size <- -log10(GO.genes.data$padj)
GO.genes.data$gene <- rownames(GO.genes.data)
GO.genes.data$treatment <- "GLP-1/MK-801"

# Plot
GO.genes.plot <- ggplot(GO.genes.data, aes(x = gene, y=treatment)) +
  geom_tile(size = 1, color = "white", fill="grey99") +
  geom_point(aes(size = size, color = log2FoldChange)) + 
  scale_size(name = expression(paste(bold(-log[10]),bold("("),
                                     bolditalic("P"),bold(")"))),
             range = c(0, 8), limits = c(0, max(GO.genes.data$size)),
             breaks = c(2, 4, 6)) +
  scale_color_gradient2(low = "#3C5488B2", mid = "white", high = "#DC0000B2",
                         breaks = c(seq(-1.5, 1.5, 0.5)),
                        name = expression(paste(bold(log[2]),bold(" fold change")))) +
  theme_pubr(legend = "top") + xlab(NULL) + ylab(NULL) + 
  theme(axis.text.x = element_text(size=10, face="bold", angle=45, hjust = 1), 
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(), 
        legend.title = element_text(size=10, face="bold"), 
        legend.text = element_text(size=10, face="bold"),
        axis.line = element_line(colour = "black"),
        # margin: top, right, bottom, and left
        plot.margin = unit(c(0, 0, 0, 1), "cm")) +
  guides(color = guide_colorbar(title.vjust = 0.8),
         shape = guide_legend(title.vjust = 0.8))
GO.genes.plot

ggsave("/projects/mludwig/Petersen-2022/output/figures/figure5b.png", 
       GO.genes.plot, height = 45, width = 180,
       units = "mm")
```
