---
title: "figure3"
author: "Mette Ludwig"
date: "2022-01-06"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---


```{r}
library(DESeq2)
library(dplyr)
library(tidyr)
library(plyr)
library(forcats)
library(ggplot2)
library(ggpubr)
library(ggrepel)
library(VennDiagram)
library(wesanderson)
library(stringr)
library(gprofiler2)
library(openxlsx)
library(rlist)
library(fgsea)
library(cowplot)
```

# Load RNA-seq data
```{r}
# Mapping data 
mapping.data <- read.table(gzfile("/projects/mludwig/Ludwig-2021/data/gene_info/Mus_musculus.GRCm38.90.gene_name_version2ensembl.txt.gz"), 
                           header = T)

# Bulk data 
path.meta <- "/projects/mludwig/Petersen-2022/data/bulk/jona-seq_meta.csv"
path.counts <- "/projects/mludwig/Petersen-2022/data/bulk/jona-seq_counts.csv"

bulk.meta <- read.table(path.meta, sep = ",", header = T, row.names = 1)
bulk.meta$index.plate.well <- bulk.meta$Index_Plate_Well 
bulk.meta$sample <- gsub("X", "hypo", bulk.meta$Sample_name)
bulk.meta$sample <- gsub("_Hy", "", bulk.meta$sample)
bulk.meta$name <- paste(bulk.meta$sample, bulk.meta$condition, sep = "_")
bulk.meta <- bulk.meta[,c("name", "sample", "condition", "index.plate.well")]

bulk.counts <- read.table(path.counts, sep = ",", header = T)

# Get gene IDs
bulk.counts[,1] <- mapping.data$gene_name_optimal[match(bulk.counts[,1], 
                                                   mapping.data$ensembl_gene_id)]
bulk.counts <- bulk.counts[!(is.na(bulk.counts[,1])),]
rownames(bulk.counts) <- bulk.counts[,1]
bulk.counts <- bulk.counts[,-1]

colnames(bulk.counts) <- gsub("X", "hypo", colnames(bulk.counts))
colnames(bulk.counts) <- gsub("_Hy", "", colnames(bulk.counts))

# Get treatment groups
colnames(bulk.counts) <- paste(colnames(bulk.counts), 
                               bulk.meta$condition[match(colnames(bulk.counts), 
                                                         bulk.meta$sample)], sep = "_")
```

# Compute differentially expressed genes
```{r}
# Peptide vs. vehicle
counts.sub <- bulk.counts[, grep("vehicle|peptide", colnames(bulk.counts))]
condition <- gsub(".*_", "", colnames(counts.sub))
condition <- factor(condition, levels = c("vehicle", "peptide"))
dds <- DESeqDataSetFromMatrix(counts.sub, DataFrame(condition), ~ condition)
dds <- DESeq(dds)
DEGs.peptide <- results(dds, name="condition_peptide_vs_vehicle") 
DEGs.peptide <- data.frame(DEGs.peptide)

# Molecule vs. vehicle
counts.sub <- bulk.counts[, grep("vehicle|molecule", colnames(bulk.counts))]
condition <- gsub(".*_", "", colnames(counts.sub))
condition <- factor(condition, levels = c("vehicle", "molecule"))
dds <- DESeqDataSetFromMatrix(counts.sub, DataFrame(condition), ~ condition)
dds <- DESeq(dds)
DEGs.molecule <- results(dds, name="condition_molecule_vs_vehicle") 
DEGs.molecule <- data.frame(DEGs.molecule)

# Conjugate vs. vehicle
counts.sub <- bulk.counts[, grep("vehicle|conjugate", colnames(bulk.counts))]
condition <- gsub(".*_", "", colnames(counts.sub))
condition <- factor(condition, levels = c("vehicle", "conjugate"))
dds <- DESeqDataSetFromMatrix(counts.sub, DataFrame(condition), ~ condition)
dds <- DESeq(dds)
DEGs.conjugate <- results(dds, name="condition_conjugate_vs_vehicle") 
DEGs.conjugate <- data.frame(DEGs.conjugate)

# Save 
DEGs.conjugate.save <- DEGs.conjugate[!(is.na(DEGs.conjugate$padj)), ]
DEGs.conjugate.save <- DEGs.conjugate.save[order(DEGs.conjugate.save$pvalue),]
DEGs.conjugate.save <- cbind(gene = rownames(DEGs.conjugate.save), DEGs.conjugate.save)
write.xlsx(DEGs.conjugate.save, 
           file = "/projects/mludwig/Petersen-2022/output/DEGs/DEGs_conjugate_vs_vehicle.xlsx",
           rowNames = F)
```


# Figure 3d
```{r}
# Construct venn diagram
DEGs.peptide.venn <- DEGs.peptide[which(DEGs.peptide$padj < 0.05),]
DEGs.molecule.venn <- DEGs.molecule[which(DEGs.molecule$padj < 0.05),]
DEGs.conjugate.venn <- DEGs.conjugate[which(DEGs.conjugate$padj < 0.05),]

genes <- union(union(rownames(DEGs.peptide.venn), rownames(DEGs.molecule.venn)), rownames(DEGs.conjugate.venn))
venn.data <- data.frame(matrix(0, nrow = length(genes), ncol = 3))
rownames(venn.data) <- genes
colnames(venn.data) <- c("peptide", "molecule", "conjugate")

venn.data$peptide[which(rownames(venn.data) %in% rownames(DEGs.peptide.venn))] <- 1
venn.data$molecule[which(rownames(venn.data) %in% rownames(DEGs.molecule.venn))] <- 1
venn.data$conjugate[which(rownames(venn.data) %in% rownames(DEGs.conjugate.venn))] <- 1

# Plot
pal <- wes_palette("Darjeeling2", 4, type = "discrete")[2:4]
grid.newpage()
png("/projects/mludwig/Petersen-2022/output/figures/figure3d.png",
    pointsize = 1,
    width = 960, height = 960, res = 2250)
draw.triple.venn(area1 = nrow(subset(venn.data, peptide == 1)), 
                 area2 = nrow(subset(venn.data, molecule == 1)),
                 area3 = nrow(subset(venn.data, conjugate == 1)), 
                 n12 = nrow(subset(venn.data, peptide == 1 & molecule == 1)), 
                 n23 = nrow(subset(venn.data, molecule == 1 & conjugate == 1)), 
                 n13 = nrow(subset(venn.data, peptide == 1 & conjugate == 1)), 
                 n123 = nrow(subset(venn.data, peptide == 1 & molecule == 1 & 
                                      conjugate == 1)), 
                 category = c("Peptide", "Molecule", "Conjugate"), 
                 lty = "blank",
                 fontfamily = "helvetica",
                 cat.fontfamily = "helvetica",
                 cat.fontface = "bold",
                 fill = c(pal[[1]], pal[[2]], pal[[3]]),
                 cat.dist = .1,
                 margin = 0.05,
                 cat.just = list(c(-1.5, 27), c(2, 27), c(0.5, -30))) 
dev.off()
```

# Figure 3e
```{r}
# Construct volcano plot
volcano <- DEGs.conjugate[!(is.na(DEGs.conjugate$padj)), ]
volcano$label <- ""
volcano$label[min_rank(volcano$padj) <= 15] <- rownames(volcano)[min_rank(volcano$padj) <= 15]
volcano$col <- 4
volcano$col[which(volcano$padj <= 0.05 & abs(volcano$log2FoldChange) >= 0.5)] <- 1
volcano$col[which(volcano$padj <= 0.05 & abs(volcano$log2FoldChange) < 0.5)] <- 2
volcano$col[which(volcano$padj > 0.05 & abs(volcano$log2FoldChange) >= 0.5)] <- 3
volcano <- data.frame(volcano[, c("log2FoldChange", "padj", "col", "label")])

# Plot
volcano.plot <- ggplot(volcano, aes(y=(-log10(padj)), x=log2FoldChange, 
                                    fill=factor(col), label=label)) +
  xlab(expression(bold(paste(Log[2], " fold-change")))) + 
  ylab(expression(paste(bold(-log[10]),bold("("), bolditalic("P"),
                                   bold(")")))) + 
  geom_point(shape = 21, size = 3, alpha = 1) + 
  geom_hline(yintercept = -log10(0.05), linetype = "dashed") +
  geom_vline(xintercept = c(-.5, .5), linetype = "dashed") + 
  geom_text_repel(fontface="bold", size=3, color = "black") + theme_pubr() + 
      theme(legend.position = "none",
            axis.title = element_text(size=10, face="bold"),
            axis.text = element_text(size=10, face="bold"))  + 
      scale_fill_manual(values = c("1" = "#960000", "2" = "#CC9999", 
                                   "3" = "grey40", "4" = "grey80")) 
volcano.plot

ggsave("/projects/mludwig/Petersen-2022/output/figures/figure3e.png", 
       volcano.plot, height = 60, width = 90,
       units = "mm")
```


# Figure 3g
```{r}
# Identify top GO terms
query <- rownames(DEGs.conjugate.venn)

module.GO.terms <- gost(query = query,
                        organism = "mmusculus",
                        ordered_query = F, correction_method = "fdr",
                        exclude_iea = T,
                        sources = c("GO:BP", "GO:MF", "GO:CC", "KEGG", "REAC"),
                        evcodes = T)

GO.terms <- module.GO.terms[[1]] %>% select(term_name, term_size, p_value,
                                            intersection_size, intersection, term_id) %>%
  filter(term_size <= 500) %>% arrange(p_value) %>% top_n(5, -p_value) %>%
  mutate(x = fct_reorder(str_wrap(term_name, 20), -p_value)) %>%
  mutate(y = -log10(p_value))

# Save
write.xlsx(GO.terms[, !(colnames(GO.terms) == "intersection")],
           file = "/projects/mludwig/Petersen-2022/output/GO_terms/GO_terms_conjugate_vs_vehicle.xlsx",
           rowNames = F)

# Plot
GO.plot <- ggplot(GO.terms, aes(x,y)) + 
  geom_col(colour="black", width=1, fill="#D69C4E", alpha=0.5, size=0.3) +
  theme_pubr(legend="none") + 
  theme(axis.text.x = element_text(size=8, face="bold"),
        axis.text.y = element_text(size=8, face="bold"),
        axis.title.x = element_text(size=8, face="bold"),
        # margin: top, right, bottom, and left
        plot.margin = unit(c(0, 0.5, 0, 0), "cm")) +
  ggsci::scale_fill_lancet() +
  coord_flip() +
  xlab(NULL) + ylab(expression(paste(bold(-log[10]),bold("("),
                                       bolditalic("P"),bold(")")))) 
GO.plot

ggsave("/projects/mludwig/Petersen-2022/output/figures/figure3g.png", 
       GO.plot, height = 42, width = 70,
       units = "mm")
```

# Figure 3f
```{r}
# Identify differentially expressed genes intersecting with top GO terms
GO.genes <- lapply(GO.terms$intersection, noquote)
GO.genes <- lapply(GO.genes, function(x) str_split(x, pattern = ","))
GO.genes <- unique(unlist(GO.genes))

ntop <- 20
top.genes <- GO.genes[order(DEGs.conjugate[GO.genes, "padj"], decreasing = F)][1:ntop]

GO.genes.data1 <- DEGs.conjugate[top.genes, ]
GO.genes.data1$gene <- rownames(GO.genes.data1)
GO.genes.data1$treatment <- "GLP-1/MK-801"

GO.genes.data2 <- DEGs.molecule[top.genes, ]
GO.genes.data2$gene <- rownames(GO.genes.data2)
GO.genes.data2$treatment <- "MK-801"


GO.genes.data3 <- DEGs.peptide[top.genes, ]
GO.genes.data3$gene <- rownames(GO.genes.data3)
GO.genes.data3$treatment <- "GLP-1"


GO.genes.data <- rbind(GO.genes.data1, GO.genes.data2, GO.genes.data3)
GO.genes.data$size <- -log10(GO.genes.data$padj)
GO.genes.data$treatment <- factor(GO.genes.data$treatment, levels = c("GLP-1", "MK-801", "GLP-1/MK-801"))

# Plot
GO.genes.plot <- ggplot(GO.genes.data, aes(x = gene, y=treatment)) +
   geom_tile(size = 1, color = "white", fill="grey99") +
  geom_point(aes(size = size, color = log2FoldChange)) + 
  scale_size(name = expression(paste(-log[10],"(",
                                     italic("P"),")")),
             range = c(0, 3), breaks = c(2, 7, 12),
             limits = c(0, max(GO.genes.data$size))) +
  scale_color_gradient2(low = "#28A1F1", mid = "white", high = "#D9393A",
                         breaks = c(seq(-1.5, 1.5, 0.5)),
                        name = expression(paste(log[2]," fold change"))) +
  theme_pubr(legend = "top") + xlab(NULL) + ylab(NULL) + 
  theme(axis.text.x = element_text(size=6, angle=45, hjust = 1), 
        axis.text.y = element_text(size=6),
        axis.ticks.x = element_line(colour = "black", size = 0.3),
        legend.title = element_text(size=5), 
        legend.text = element_text(size=5),
        legend.spacing.x = unit(-0.2, "mm"),
        legend.key.width=unit(5, "mm"),
        axis.line = element_line(colour = "black", size = 0.3),
        # margin: top, right, bottom, and left
        plot.margin = unit(c(0, 0, 0, 0), "cm"),
        legend.box.margin = unit(c(0, 0, 0, 0), "cm"),
        text=element_text(family="sans")) +
    guides(color = guide_colorbar(title.vjust = 0.7, order = 2),
           size = guide_legend(title.vjust = 0.5, order = 1))

plot1 <- plot_grid(get_legend(GO.genes.plot),
                   GO.genes.plot + theme(legend.position = "none"),
                   align = "v", axis = "lr", ncol = 1, 
                   rel_heights = c(2, 5))

ggsave("/projects/mludwig/Petersen-2022/output/figures/figure3f.pdf",        
       plot1, height = 1.5, width = 2.66, units = "in", dpi = 1500)
```
